specification/requirements.adoc

```diff
@@ -1,5 +1,167 @@
 [[requirements]]
 == Requirements
 
-Hardware (platform and device) and software (host and guest) requirements to
-support CoVE-IO.
+In order to extend TVM TCBs with external and untrusted devices, the host
+platform and software stack, together with assigned devices, must meet a
+specific set of requirements.
+
+=== Device
+
+CoVE-IO compliant implementations support extending TVMs TCBs with devices if
+and only if they meet the following requirements:
+
+==== PCI
+
+A CoVE-IO supported device must be a PCIe device with optional SR-IOV
+capabilities. When supporting SR-IOV, the device can be split into multiple
+Virtual Functions (VFs). A TVM assigned interface can be the whole device, also
+known as the Physiscal Device (PF) or any of its VFs. In both cases, this unit
+of assignment is referred to as a TEE I/O Device Interface (TDI).
+
+A host enumerated TDI must provide the following PCIe capabilities:
+
+===== TEE Device Interface Protocol (TDISP)
+
+In order to interoperate with a RISC-V CoVE-IO implementation, PCIe devices must
+support the <<TDISP>> protocol, as defined in the latest TDISP ECN.
+
+In particular, CoVE-IO compatible device firmwares must run a Device Security
+Manager (DSM) to enforce the TDISP defined security attributes and policies.
+The DSM must support both full PCIe devices (PFs) and Virtual Functions (VFs)
+assignment to TVMs.
+
+TODO: List of required TDISP messages
```

Comment:
I believe TDISP spec already defines the requirement.

Do we want to list **any additional requirement** here?
Or we just say: The device must comply with TDISP. That is enough?